Fix #6022 Avoid MultilineHashBraceLayout auto-correct syntax error

bacchir opened the pull request:
See [#6022](https://github.com/rubocop-hq/rubocop/issues/6022)

The same issue happens with `Layout/MultilineArrayBraceLayout` and it's also fixed by this PR. My approach here is to not auto-correct in case there's a comment after the last element of the hash/array and the hash/array is inside a method call.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

bacchir commented:
I'll spend some time now refactoring the multiline literal braces shared examples because they are starting to get messy IMO.

bbatsov commented on spec/support/multiline_literal_brace_layout_examples.rb:
```diff
@@ -330,6 +330,44 @@ def construct(*args)
     end
 
+    if [
+      RuboCop::Cop::Layout::MultilineArrayBraceLayout,
```
This seems pretty odd. Probably we just need different contexts for both or something like this.

bacchir replied:
You are right, I'm going to refactor these shared examples

bacchir replied:
@bbatsov I just refactored the shared examples in order to avoid ifs. Please take a look and let me know if it's ok now.

bbatsov replied:
Hmm, I still see this `if` in the diff. :-)

bacchir replied:
You are totally right, sorry. I'll fix it.

bacchir replied:
Sorry man, now it's correct.

bacchir commented:
@bbatsov I've fixed up this PR and I believe it's now ready to get merged. Any comments?

bacchir commented:
Rebased master and fixed failing tests related to [this change](https://github.com/rubocop-hq/rubocop/commit/82a005d4f2244bf987e2bde8823b6986b29be53b)

bbatsov commented:
👍
